### PR TITLE
Continue Enet PR

### LINKED ISSUE #947: IPX tunnel enhancement
One of the issues that has persisted in DosBox for many years is the usability of the IPX tunnel over internet connections.  This feature works really well over LAN connections but suffers greatly over the Internet.  The likely issues have to do with higher latency, packet loss,  and UDP packets arriving out of order.  One thing to keep in mind is that back in the 1990s there have been services that would tunnel IPX over a dialup connection on a BBS or other service and it worked rather well.  These connections had even higher latency and packet loss issues than current broadband.

Recently, there was a discussion about this on the Discord server, and a user over there, granminigun, has suggested the use of the UDP library [enet](https://github.com/zpl-c/enet).  A quick review shows some promise to address some of the main problems that are theorized to cause issues.

Some of the issues I have noticed with the current IPX tunnel implementation:
- Doom - Players get stuck to wall, "rubber band" effect
- Rise of the Triad - Will not complete a connection at all

These issues may vary from person to person and depend on how the game handles the IPX connection.

What I propose is the following:
- Wrap the current IPX UDP connection within enet
- Provide a fallback option to the original protocol for compatibility with upstream and other forks, as well as for LAN connections that don't need any help from the library

These are currently my thoughts and there may be even better ways to handle this as well.  I appreciate all feedback.

### PR DESCRIPTION
**Pull request by kcgen**
Thanks to:
 - @sduensin, for the baseline PR (https://github.com/dosbox-staging/dosbox-staging/pull/1398)
 - @GranMinigun, for adding more help and reformatting all commits (https://github.com/GranMinigun/dosbox-staging/tree/gm/serial-enet-cleanup-1).

**Note**: Enet introduce compiler warnings and PVS Studio issues, which have been permitted in this PR. They've been [submitted upstream](https://github.com/zpl-c/enet/issues/35) and are [being tracked here](https://github.com/dosbox-staging/dosbox-staging/issues/1442).

Fixes #947.

### CODE REVIEW

File: src/dos/dos_programs.cpp [hunk 1]
```diff
@@ -496,6 +497,43 @@ void DOS_SetupPrograms(void)
...
+	        "  For MODEM: IRQ, LISTENPORT, SOCK\n"
+	        "  For NULLMODEM: IRQ, SERVER, RXDELAY, TXDELAY, TELNET, USEDTR,\n"
+	        "                 TRANSPARENT, PORT, INHSOCKET, SOCK\n"
+	        "\n"
+	        "Examples:\n"
+	        "  SERIAL /l\n"
+	        "  SERIAL 1 DISABLED\n"
+	        "  SERIAL 2 MODEM LISTENPORT:5000 SOCK:1\n"
+	        "  SERIAL 1 DIRECTSERIAL REALPORT:COM1\n"
+	        "  SERIAL 1 DIRECTSERIAL REALPORT:tty50\n");
```
Comment (GranMinigun) [inline src/dos/dos_programs.cpp:530]: This really should be somehow shortened. Currently it doesn't fit into default 80x25 window.

Comment (Grounded0) [reply src/dos/dos_programs.cpp:530]: Can you take a screenshot of it in a 80x25 window so we have a better idea how to make it more consise?

Comment (GranMinigun) [reply src/dos/dos_programs.cpp:530]: Sure.
Screenshot_20211215_091825

Comment (Grounded0) [reply src/dos/dos_programs.cpp:530]: Can you make one in 80x43 too so we can see the whole text.

Comment (Grounded0) [reply src/dos/dos_programs.cpp:530]: That's `MODE 80,43` with FreeDOS.

Comment (GranMinigun) [reply src/dos/dos_programs.cpp:530]: Screenshot_20211217_105747

**Comment by Grounded0** [comment]
Usage: <eliminate \n here

Port # (required): <eliminate \n here> 

Type (required): <eliminate \n here>

Examples: <eliminate \n here>

Eliminate line: SERIAL 1 DIRECTSERIAL REALPORT:tty50

File: src/dos/dos_programs.cpp (hunk 1 above)
Comment (Grounded0) [reply src/dos/dos_programs.cpp:530]: Usage: <eliminate \n here

Port # (required): <eliminate \n here>

Type (required): <eliminate \n here>

Examples: <eliminate \n here>

Eliminate line: SERIAL 1 DIRECTSERIAL REALPORT:tty50

**Comment by kcgen** [comment]
Note: the two LGTM alerts are coming from `enet.h`.

**Pull request merged**